main/src/library/Fixpoint/Interpreter.flix

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2021 Benjamin Dahse
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use Fixpoint/Ram.{RamStmt, RelOp, RelSym, RamTerm, BoolExp, RowVar, toDenotation};
+use Fixpoint/Ast.Denotation;
+
+instance Order[Array[v]] with Order[v] {
+    pub def compare(a: Array[v], b: Array[v]): Comparison = compareHelper(a, b, 0) as & Pure
+}
+
+def compareHelper(a: Array[v], b: Array[v], i: Int32): Comparison & Impure with Order[v] =
```

Comment:
I am not sure I entirely understand the ordering here. Is it supposed to be pairwise lexicographic? 

I would have expected one call to `Array.length(b)`. (Mind you, my previous comment here was that it is not necessary to call Array.length for every iteration.)

Reply:
Yes it's pairwise lexicographic but I'm "cheating" because I know `a` and `b` have the same length (in our case).